src/Symfony/Component/Messenger/QueryBusTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+use Symfony\Component\Messenger\Exception\LogicException;
+use Symfony\Component\Messenger\Middleware\HandleMessageMiddleware;
+use Symfony\Component\Messenger\Stamp\HandledStamp;
+
+/**
+ * Leverages a message bus to expect a single, synchronous message handling and return its result.
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @experimental in 4.3
+ */
+trait QueryBusTrait
+{
+    /**
+     * Dispatches the given message and returns the result from the handler returned value.
+     *
+     * @param object|Envelope $message The message or the message pre-wrapped in an envelope
+     *
+     * @return mixed The handler returned value
+     */
+    public function dispatchQuery(MessageBusInterface $messageBus, $message)
```

Comment:
private function dispatchQuery($message) + private $queryBus property that the consumer would need to set in their constructor (eg type-hinting `MessageBusInterface $queryBus`)